File: apiwdbe/WdbeQPrsList.cpp
/**
	* \file WdbeQPrsList.cpp
	* API code for table TblWdbeQPrsList (implementation)
	* \author Alexander Wirthmueller
	* \date created: 23 Aug 2020
	* \date modified: 23 Aug 2020
	*/

#include "WdbeQPrsList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQPrsList
 ******************************************************************************/

WdbeQPrsList::WdbeQPrsList(
			const uint jnum
			, const string stubGrp
			, const string stubOwn
			, const string Title
			, const string Firstname
			, const string Lastname
			, const string srefIxVSex
			, const string titIxVSex
			, const string telVal
			, const string emlVal
		) {
	this->jnum = jnum;
	this->stubGrp = stubGrp;
	this->stubOwn = stubOwn;
	this->Title = Title;
	this->Firstname = Firstname;
	this->Lastname = Lastname;
	this->srefIxVSex = srefIxVSex;
	this->titIxVSex = titIxVSex;
	this->telVal = telVal;
	this->emlVal = emlVal;
};

bool WdbeQPrsList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQPrsList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "stubGrp", "grp", stubGrp);
		extractStringUclc(docctx, basexpath, "stubOwn", "own", stubOwn);
		extractStringUclc(docctx, basexpath, "Title", "tit", Title);
		extractStringUclc(docctx, basexpath, "Firstname", "fnm", Firstname);
		extractStringUclc(docctx, basexpath, "Lastname", "lnm", Lastname);
		extractStringUclc(docctx, basexpath, "srefIxVSex", "sex", srefIxVSex);
		extractStringUclc(docctx, basexpath, "titIxVSex", "sex2", titIxVSex);
		extractStringUclc(docctx, basexpath, "telVal", "tel", telVal);
		extractStringUclc(docctx, basexpath, "emlVal", "eml", emlVal);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQPrsList
 ******************************************************************************/

ListWdbeQPrsList::ListWdbeQPrsList() {
};

ListWdbeQPrsList::ListWdbeQPrsList(
			const ListWdbeQPrsList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQPrsList(*(src.nodes[i]));
};

ListWdbeQPrsList& ListWdbeQPrsList::operator=(
			const ListWdbeQPrsList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQPrsList(*(src.nodes[i]));

	return *this;
};

ListWdbeQPrsList::~ListWdbeQPrsList() {
	clear();
};

void ListWdbeQPrsList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQPrsList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQPrsList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQPrsList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQPrsList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQPrsList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQPrsList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwdbe/WdbeQPrsList.java
/**
	* \file WdbeQPrsList.h
	* Java API code for record of table TblWdbeQPrsList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQPrsList {
	
	public WdbeQPrsList(
				int jnum
				, String stubGrp
				, String stubOwn
				, String Title
				, String Firstname
				, String Lastname
				, String srefIxVSex
				, String titIxVSex
				, String telVal
				, String emlVal
			) {
		this.jnum = jnum;
		this.stubGrp = stubGrp;
		this.stubOwn = stubOwn;
		this.Title = Title;
		this.Firstname = Firstname;
		this.Lastname = Lastname;
		this.srefIxVSex = srefIxVSex;
		this.titIxVSex = titIxVSex;
		this.telVal = telVal;
		this.emlVal = emlVal;
	};

	public int jnum;
	public String stubGrp;
	public String stubOwn;
	public String Title;
	public String Firstname;
	public String Lastname;
	public String srefIxVSex;
	public String titIxVSex;
	public String telVal;
	public String emlVal;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQPrsList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			stubGrp = Xmlio.extractStringUclc(doc, basexpath, "stubGrp", "grp", null, 0);
			stubOwn = Xmlio.extractStringUclc(doc, basexpath, "stubOwn", "own", null, 0);
			Title = Xmlio.extractStringUclc(doc, basexpath, "Title", "tit", null, 0);
			Firstname = Xmlio.extractStringUclc(doc, basexpath, "Firstname", "fnm", null, 0);
			Lastname = Xmlio.extractStringUclc(doc, basexpath, "Lastname", "lnm", null, 0);
			srefIxVSex = Xmlio.extractStringUclc(doc, basexpath, "srefIxVSex", "sex", null, 0);
			titIxVSex = Xmlio.extractStringUclc(doc, basexpath, "titIxVSex", "sex2", null, 0);
			telVal = Xmlio.extractStringUclc(doc, basexpath, "telVal", "tel", null, 0);
			emlVal = Xmlio.extractStringUclc(doc, basexpath, "emlVal", "eml", null, 0);
			
			return true;
		};

		return false;
	};
};

